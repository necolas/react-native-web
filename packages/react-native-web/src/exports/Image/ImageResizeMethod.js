/**
 * @flow
 */

const ImageResizeMethod = {
  auto: 'auto',
  resize: 'resize',
  scale: 'scale'
};

export default ImageResizeMethod;
