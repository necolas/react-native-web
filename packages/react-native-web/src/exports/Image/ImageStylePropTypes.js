/**
 * @flow
 */

import BorderPropTypes from '../../modules/BorderPropTypes';
import ColorPropType from '../ColorPropType';
import ImageResizeMode from './ImageResizeMode';
import LayoutPropTypes from '../../modules/LayoutPropTypes';
import ShadowPropTypes from '../../modules/ShadowPropTypes';
import TransformPropTypes from '../../modules/TransformPropTypes';
import { number, oneOf, string } from 'prop-types';

const ImageStylePropTypes = {
  ...BorderPropTypes,
  ...LayoutPropTypes,
  ...ShadowPropTypes,
  ...TransformPropTypes,
  backgroundColor: ColorPropType,
  opacity: number,
  resizeMode: oneOf(Object.keys(ImageResizeMode)),
  /**
   * @platform unsupported
   */
  overlayColor: string,
  tintColor: ColorPropType,
  /**
   * @platform web
   */
  boxShadow: string
};

export default ImageStylePropTypes;
