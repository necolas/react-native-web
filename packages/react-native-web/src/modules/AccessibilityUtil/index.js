/**
 * Copyright (c) Nicolas Gallagher.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow
 */

import isDisabled from './isDisabled';
import propsToAccessibilityComponent from './propsToAccessibilityComponent';
import propsToAriaRole from './propsToAriaRole';

const AccessibilityUtil = {
  isDisabled,
  propsToAccessibilityComponent,
  propsToAriaRole
};

export default AccessibilityUtil;
