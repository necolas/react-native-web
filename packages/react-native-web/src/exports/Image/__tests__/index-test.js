/**
 * Copyright (c) Nicolas Gallagher.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

/* eslint-disable react/jsx-no-bind */

import { act } from 'react-dom/test-utils';
import * as AssetRegistry from '../../../modules/AssetRegistry';
import Image from '../';
import ImageLoader, { ImageUriCache } from '../../../modules/ImageLoader';
import PixelRatio from '../../PixelRatio';
import React from 'react';
import { render } from '@testing-library/react';

const originalImage = window.Image;

describe('components/Image', () => {
  beforeEach(() => {
    ImageUriCache._entries = {};
    window.Image = jest.fn(() => ({}));
  });

  afterEach(() => {
    window.Image = originalImage;
  });

  test('prop "accessibilityLabel"', () => {
    const defaultSource = { uri: 'https://google.com/favicon.ico' };
    const { container } = render(
      <Image
        accessibilityLabel="accessibilityLabel"
        defaultSource={defaultSource}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('prop "blurRadius"', () => {
    const defaultSource = { uri: 'https://google.com/favicon.ico' };
    const { container } = render(
      <Image blurRadius={5} defaultSource={defaultSource} />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  describe('prop "defaultSource"', () => {
    test('sets background image when value is an object', () => {
      const defaultSource = { uri: 'https://google.com/favicon.ico' };
      const { container } = render(<Image defaultSource={defaultSource} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    test('sets background image when value is a string', () => {
      // emulate require-ed asset
      const defaultSource = 'https://google.com/favicon.ico';
      const { container } = render(<Image defaultSource={defaultSource} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    test('sets "height" and "width" styles if missing', () => {
      const defaultSource = {
        uri: 'https://google.com/favicon.ico',
        height: 10,
        width: 20
      };
      const { container } = render(<Image defaultSource={defaultSource} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    test('does not override "height" and "width" styles', () => {
      const defaultSource = {
        uri: 'https://google.com/favicon.ico',
        height: 10,
        width: 20
      };
      const { container } = render(
        <Image
          defaultSource={defaultSource}
          style={{ height: 20, width: 40 }}
        />
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  test('prop "draggable"', () => {
    const defaultSource = { uri: 'https://google.com/favicon.ico' };
    const { container } = render(
      <Image defaultSource={defaultSource} draggable={true} />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('prop "focusable"', () => {
    const { container } = render(<Image focusable={true} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('prop "nativeID"', () => {
    const { container } = render(<Image nativeID="nativeID" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  describe('prop "onLoad"', () => {
    test('is called after image is loaded from network', () => {
      jest.useFakeTimers();
      ImageLoader.load = jest.fn().mockImplementation((_, onLoad, onError) => {
        onLoad();
      });
      const onLoadStartStub = jest.fn();
      const onLoadStub = jest.fn();
      const onLoadEndStub = jest.fn();
      render(
        <Image
          onLoad={onLoadStub}
          onLoadEnd={onLoadEndStub}
          onLoadStart={onLoadStartStub}
          source="https://test.com/img.jpg"
        />
      );
      jest.runOnlyPendingTimers();
      expect(onLoadStub).toBeCalled();
    });

    test('is called after image is loaded from cache', () => {
      jest.useFakeTimers();
      ImageLoader.load = jest.fn().mockImplementation((_, onLoad, onError) => {
        onLoad();
      });
      const onLoadStartStub = jest.fn();
      const onLoadStub = jest.fn();
      const onLoadEndStub = jest.fn();
      const uri = 'https://test.com/img.jpg';
      ImageUriCache.add(uri);
      render(
        <Image
          onLoad={onLoadStub}
          onLoadEnd={onLoadEndStub}
          onLoadStart={onLoadStartStub}
          source={uri}
        />
      );
      jest.runOnlyPendingTimers();
      expect(onLoadStub).toBeCalled();
      ImageUriCache.remove(uri);
    });

    test('is called on update if "uri" is different', () => {
      const onLoadStartStub = jest.fn();
      const onLoadStub = jest.fn();
      const onLoadEndStub = jest.fn();
      const { rerender } = render(
        <Image
          onLoad={onLoadStub}
          onLoadEnd={onLoadEndStub}
          onLoadStart={onLoadStartStub}
          source={'https://test.com/img.jpg'}
        />
      );
      act(() => {
        rerender(
          <Image
            onLoad={onLoadStub}
            onLoadEnd={onLoadEndStub}
            onLoadStart={onLoadStartStub}
            source={'https://blah.com/img.png'}
          />
        );
      });
      expect(onLoadStub.mock.calls.length).toBe(2);
      expect(onLoadEndStub.mock.calls.length).toBe(2);
    });

    test('is not called on update if "uri" is the same', () => {
      const onLoadStartStub = jest.fn();
      const onLoadStub = jest.fn();
      const onLoadEndStub = jest.fn();
      const { rerender } = render(
        <Image
          onLoad={onLoadStub}
          onLoadEnd={onLoadEndStub}
          onLoadStart={onLoadStartStub}
          source={'https://test.com/img.jpg'}
        />
      );
      act(() => {
        rerender(
          <Image
            onLoad={onLoadStub}
            onLoadEnd={onLoadEndStub}
            onLoadStart={onLoadStartStub}
            source={'https://test.com/img.jpg'}
          />
        );
      });
      expect(onLoadStub.mock.calls.length).toBe(1);
      expect(onLoadEndStub.mock.calls.length).toBe(1);
    });

    test('is not called on update if "uri" is the same and given as an object', () => {
      const onLoadStartStub = jest.fn();
      const onLoadStub = jest.fn();
      const onLoadEndStub = jest.fn();
      const { rerender } = render(
        <Image
          onLoad={onLoadStub}
          onLoadEnd={onLoadEndStub}
          onLoadStart={onLoadStartStub}
          source={{ uri: 'https://test.com/img.jpg', width: 1, height: 1 }}
        />
      );
      act(() => {
        rerender(
          <Image
            onLoad={onLoadStub}
            onLoadEnd={onLoadEndStub}
            onLoadStart={onLoadStartStub}
            source={{ uri: 'https://test.com/img.jpg', width: 1, height: 1 }}
          />
        );
      });
      expect(onLoadStub.mock.calls.length).toBe(1);
      expect(onLoadEndStub.mock.calls.length).toBe(1);
    });
  });

  describe('prop "resizeMode"', () => {
    ['contain', 'cover', 'none', 'repeat', 'stretch', undefined].forEach(
      (resizeMode) => {
        test(`value "${resizeMode}"`, () => {
          const { container } = render(<Image resizeMode={resizeMode} />);
          expect(container.firstChild).toMatchSnapshot();
        });
      }
    );
  });

  describe('prop "source"', () => {
    test('does not throw', () => {
      const sources = [
        null,
        '',
        {},
        { uri: '' },
        { uri: 'https://google.com' }
      ];
      sources.forEach((source) => {
        expect(() => render(<Image source={source} />)).not.toThrow();
      });
    });

    test('is not set immediately if the image has not already been loaded', () => {
      const uri = 'https://google.com/favicon.ico';
      const source = { uri };
      const { container } = render(<Image source={source} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    test('is set immediately if the image was preloaded', () => {
      const uri = 'https://yahoo.com/favicon.ico';
      ImageLoader.load = jest
        .fn()
        .mockImplementationOnce((_, onLoad, onError) => {
          onLoad();
        });
      return Image.prefetch(uri).then(() => {
        const source = { uri };
        const { container } = render(<Image source={source} />, {
          disableLifecycleMethods: true
        });
        expect(container.firstChild).toMatchSnapshot();
        ImageUriCache.remove(uri);
      });
    });

    test('is set immediately if the image has already been loaded', () => {
      const uriOne = 'https://google.com/favicon.ico';
      const uriTwo = 'https://twitter.com/favicon.ico';
      ImageUriCache.add(uriOne);
      ImageUriCache.add(uriTwo);

      // initial render
      const { container, rerender } = render(
        <Image source={{ uri: uriOne }} />
      );
      ImageUriCache.remove(uriOne);
      expect(container.firstChild).toMatchSnapshot();
      // props update
      act(() => {
        rerender(<Image source={{ uri: uriTwo }} />);
        ImageUriCache.remove(uriTwo);
      });
      expect(container.firstChild).toMatchSnapshot();
    });

    test('is correctly updated when missing in initial render', () => {
      const uri = 'https://testing.com/img.jpg';
      const { container, rerender } = render(<Image />);
      act(() => {
        rerender(<Image source={{ uri }} />);
      });
      expect(container.firstChild).toMatchSnapshot();
    });

    test('is correctly updated only when loaded if defaultSource provided', () => {
      const defaultUri = 'https://testing.com/preview.jpg';
      const uri = 'https://testing.com/fullSize.jpg';
      let loadCallback;
      ImageLoader.load = jest
        .fn()
        .mockImplementationOnce((_, onLoad, onError) => {
          loadCallback = onLoad;
        });
      const { container } = render(
        <Image defaultSource={{ uri: defaultUri }} source={{ uri }} />
      );
      expect(container.firstChild).toMatchSnapshot();
      act(() => {
        loadCallback();
      });
      expect(container.firstChild).toMatchSnapshot();
    });

    test('it correctly selects the source scale', () => {
      AssetRegistry.getAssetByID = jest.fn(() => ({
        httpServerLocation: 'static',
        name: 'img',
        scales: [1, 2, 3],
        type: 'png'
      }));

      PixelRatio.get = jest.fn(() => 1.0);
      let { container } = render(<Image source={1} />);
      expect(container.querySelector('img').src).toBe(
        'http://localhost/static/img.png'
      );

      act(() => {
        PixelRatio.get = jest.fn(() => 2.2);
        ({ container } = render(<Image source={1} />));
      });
      expect(container.querySelector('img').src).toBe(
        'http://localhost/static/img@2x.png'
      );
    });
  });

  describe('prop "style"', () => {
    test('supports "resizeMode" property', () => {
      const { container } = render(<Image style={{ resizeMode: 'contain' }} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    test('supports "shadow" properties (convert to filter)', () => {
      const { container } = render(
        <Image
          style={{ shadowColor: 'red', shadowOffset: { width: 1, height: 1 } }}
        />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    test('supports "tintcolor" property (convert to filter)', () => {
      const defaultSource = { uri: 'https://google.com/favicon.ico' };
      const { container } = render(
        <Image defaultSource={defaultSource} style={{ tintColor: 'red' }} />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    test('removes other unsupported View styles', () => {
      const { container } = render(
        <Image style={{ overlayColor: 'red', tintColor: 'blue' }} />
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  test('prop "testID"', () => {
    const { container } = render(<Image testID="testID" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('queryCache', () => {
    const uriOne = 'https://google.com/favicon.ico';
    const uriTwo = 'https://twitter.com/favicon.ico';
    ImageUriCache.add(uriOne);
    ImageUriCache.add(uriTwo);
    return Image.queryCache([uriOne, uriTwo, 'oops']).then((res) => {
      expect(res).toEqual({
        [uriOne]: 'disk/memory',
        [uriTwo]: 'disk/memory'
      });
    });
  });
});
