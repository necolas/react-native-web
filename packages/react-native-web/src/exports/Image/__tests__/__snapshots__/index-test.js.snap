// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Image prop "accessibilityLabel" 1`] = `
<div
  aria-label="accessibilityLabel"
  class="css-view-175oi2r r-flexBasis-1mlwlqe r-overflow-1udh08x r-zIndex-417010"
>
  <div
    class="css-view-175oi2r r-backgroundColor-1niwhzg r-backgroundPosition-vvn4in r-backgroundRepeat-u6sd8q r-bottom-1p0dtai r-height-1pi2tsx r-left-1d2f490 r-position-u8s1d r-right-zchlnj r-top-ipm5af r-width-13qz1uu r-zIndex-1wyyakw r-backgroundSize-4gszlv"
    style="background-image: url(https://google.com/favicon.ico);"
  />
  <img
    alt="accessibilityLabel"
    class="css-accessibilityImage-9pa8cd"
    draggable="false"
    src="https://google.com/favicon.ico"
  />
</div>
`;

exports[`components/Image prop "blurRadius" 1`] = `
<div
  class="css-view-175oi2r r-flexBasis-1mlwlqe r-overflow-1udh08x r-zIndex-417010"
>
  <div
    class="css-view-175oi2r r-backgroundColor-1niwhzg r-backgroundPosition-vvn4in r-backgroundRepeat-u6sd8q r-bottom-1p0dtai r-height-1pi2tsx r-left-1d2f490 r-position-u8s1d r-right-zchlnj r-top-ipm5af r-width-13qz1uu r-zIndex-1wyyakw r-backgroundSize-4gszlv"
    style="background-image: url(https://google.com/favicon.ico); filter: blur(5px);"
  />
  <img
    alt=""
    class="css-accessibilityImage-9pa8cd"
    draggable="false"
    src="https://google.com/favicon.ico"
  />
</div>
`;

exports[`components/Image prop "defaultSource" does not override "height" and "width" styles 1`] = `
<div
  class="css-view-175oi2r r-flexBasis-1mlwlqe r-overflow-1udh08x r-zIndex-417010"
  style="height: 20px; width: 40px;"
>
  <div
    class="css-view-175oi2r r-backgroundColor-1niwhzg r-backgroundPosition-vvn4in r-backgroundRepeat-u6sd8q r-bottom-1p0dtai r-height-1pi2tsx r-left-1d2f490 r-position-u8s1d r-right-zchlnj r-top-ipm5af r-width-13qz1uu r-zIndex-1wyyakw r-backgroundSize-4gszlv"
    style="background-image: url(https://google.com/favicon.ico);"
  />
  <img
    alt=""
    class="css-accessibilityImage-9pa8cd"
    draggable="false"
    src="https://google.com/favicon.ico"
  />
</div>
`;

exports[`components/Image prop "defaultSource" sets "height" and "width" styles if missing 1`] = `
<div
  class="css-view-175oi2r r-flexBasis-1mlwlqe r-overflow-1udh08x r-zIndex-417010"
  style="height: 10px; width: 20px;"
>
  <div
    class="css-view-175oi2r r-backgroundColor-1niwhzg r-backgroundPosition-vvn4in r-backgroundRepeat-u6sd8q r-bottom-1p0dtai r-height-1pi2tsx r-left-1d2f490 r-position-u8s1d r-right-zchlnj r-top-ipm5af r-width-13qz1uu r-zIndex-1wyyakw r-backgroundSize-4gszlv"
    style="background-image: url(https://google.com/favicon.ico);"
  />
  <img
    alt=""
    class="css-accessibilityImage-9pa8cd"
    draggable="false"
    src="https://google.com/favicon.ico"
  />
</div>
`;

exports[`components/Image prop "defaultSource" sets background image when value is a string 1`] = `
<div
  class="css-view-175oi2r r-flexBasis-1mlwlqe r-overflow-1udh08x r-zIndex-417010"
>
  <div
    class="css-view-175oi2r r-backgroundColor-1niwhzg r-backgroundPosition-vvn4in r-backgroundRepeat-u6sd8q r-bottom-1p0dtai r-height-1pi2tsx r-left-1d2f490 r-position-u8s1d r-right-zchlnj r-top-ipm5af r-width-13qz1uu r-zIndex-1wyyakw r-backgroundSize-4gszlv"
    style="background-image: url(https://google.com/favicon.ico);"
  />
  <img
    alt=""
    class="css-accessibilityImage-9pa8cd"
    draggable="false"
    src="https://google.com/favicon.ico"
  />
</div>
`;

exports[`components/Image prop "defaultSource" sets background image when value is an object 1`] = `
<div
  class="css-view-175oi2r r-flexBasis-1mlwlqe r-overflow-1udh08x r-zIndex-417010"
>
  <div
    class="css-view-175oi2r r-backgroundColor-1niwhzg r-backgroundPosition-vvn4in r-backgroundRepeat-u6sd8q r-bottom-1p0dtai r-height-1pi2tsx r-left-1d2f490 r-position-u8s1d r-right-zchlnj r-top-ipm5af r-width-13qz1uu r-zIndex-1wyyakw r-backgroundSize-4gszlv"
    style="background-image: url(https://google.com/favicon.ico);"
  />
  <img
    alt=""
    class="css-accessibilityImage-9pa8cd"
    draggable="false"
    src="https://google.com/favicon.ico"
  />
</div>
`;

exports[`components/Image prop "draggable" 1`] = `
<div
  class="css-view-175oi2r r-flexBasis-1mlwlqe r-overflow-1udh08x r-zIndex-417010"
>
  <div
    class="css-view-175oi2r r-backgroundColor-1niwhzg r-backgroundPosition-vvn4in r-backgroundRepeat-u6sd8q r-bottom-1p0dtai r-height-1pi2tsx r-left-1d2f490 r-position-u8s1d r-right-zchlnj r-top-ipm5af r-width-13qz1uu r-zIndex-1wyyakw r-backgroundSize-4gszlv"
    style="background-image: url(https://google.com/favicon.ico);"
  />
  <img
    alt=""
    class="css-accessibilityImage-9pa8cd"
    draggable="true"
    src="https://google.com/favicon.ico"
  />
</div>
`;

exports[`components/Image prop "focusable" 1`] = `
<div
  class="css-view-175oi2r r-flexBasis-1mlwlqe r-overflow-1udh08x r-zIndex-417010"
  tabindex="0"
>
  <div
    class="css-view-175oi2r r-backgroundColor-1niwhzg r-backgroundPosition-vvn4in r-backgroundRepeat-u6sd8q r-bottom-1p0dtai r-height-1pi2tsx r-left-1d2f490 r-position-u8s1d r-right-zchlnj r-top-ipm5af r-width-13qz1uu r-zIndex-1wyyakw r-backgroundSize-4gszlv"
  />
</div>
`;

exports[`components/Image prop "nativeID" 1`] = `
<div
  class="css-view-175oi2r r-flexBasis-1mlwlqe r-overflow-1udh08x r-zIndex-417010"
  id="nativeID"
>
  <div
    class="css-view-175oi2r r-backgroundColor-1niwhzg r-backgroundPosition-vvn4in r-backgroundRepeat-u6sd8q r-bottom-1p0dtai r-height-1pi2tsx r-left-1d2f490 r-position-u8s1d r-right-zchlnj r-top-ipm5af r-width-13qz1uu r-zIndex-1wyyakw r-backgroundSize-4gszlv"
  />
</div>
`;

exports[`components/Image prop "resizeMode" value "contain" 1`] = `
<div
  class="css-view-175oi2r r-flexBasis-1mlwlqe r-overflow-1udh08x r-zIndex-417010"
>
  <div
    class="css-view-175oi2r r-backgroundColor-1niwhzg r-backgroundPosition-vvn4in r-backgroundRepeat-u6sd8q r-bottom-1p0dtai r-height-1pi2tsx r-left-1d2f490 r-position-u8s1d r-right-zchlnj r-top-ipm5af r-width-13qz1uu r-zIndex-1wyyakw r-backgroundSize-ehq7j7"
  />
</div>
`;

exports[`components/Image prop "resizeMode" value "cover" 1`] = `
<div
  class="css-view-175oi2r r-flexBasis-1mlwlqe r-overflow-1udh08x r-zIndex-417010"
>
  <div
    class="css-view-175oi2r r-backgroundColor-1niwhzg r-backgroundPosition-vvn4in r-backgroundRepeat-u6sd8q r-bottom-1p0dtai r-height-1pi2tsx r-left-1d2f490 r-position-u8s1d r-right-zchlnj r-top-ipm5af r-width-13qz1uu r-zIndex-1wyyakw r-backgroundSize-4gszlv"
  />
</div>
`;

exports[`components/Image prop "resizeMode" value "none" 1`] = `
<div
  class="css-view-175oi2r r-flexBasis-1mlwlqe r-overflow-1udh08x r-zIndex-417010"
>
  <div
    class="css-view-175oi2r r-backgroundColor-1niwhzg r-backgroundRepeat-u6sd8q r-bottom-1p0dtai r-height-1pi2tsx r-left-1d2f490 r-position-u8s1d r-right-zchlnj r-top-ipm5af r-width-13qz1uu r-zIndex-1wyyakw r-backgroundPosition-1mrlafo r-backgroundSize-1sxrcry"
  />
</div>
`;

exports[`components/Image prop "resizeMode" value "repeat" 1`] = `
<div
  class="css-view-175oi2r r-flexBasis-1mlwlqe r-overflow-1udh08x r-zIndex-417010"
>
  <div
    class="css-view-175oi2r r-backgroundColor-1niwhzg r-bottom-1p0dtai r-height-1pi2tsx r-left-1d2f490 r-position-u8s1d r-right-zchlnj r-top-ipm5af r-width-13qz1uu r-zIndex-1wyyakw r-backgroundPosition-1mrlafo r-backgroundRepeat-17leim2 r-backgroundSize-1sxrcry"
  />
</div>
`;

exports[`components/Image prop "resizeMode" value "stretch" 1`] = `
<div
  class="css-view-175oi2r r-flexBasis-1mlwlqe r-overflow-1udh08x r-zIndex-417010"
>
  <div
    class="css-view-175oi2r r-backgroundColor-1niwhzg r-backgroundPosition-vvn4in r-backgroundRepeat-u6sd8q r-bottom-1p0dtai r-height-1pi2tsx r-left-1d2f490 r-position-u8s1d r-right-zchlnj r-top-ipm5af r-width-13qz1uu r-zIndex-1wyyakw r-backgroundSize-x3cy2q"
  />
</div>
`;

exports[`components/Image prop "resizeMode" value "undefined" 1`] = `
<div
  class="css-view-175oi2r r-flexBasis-1mlwlqe r-overflow-1udh08x r-zIndex-417010"
>
  <div
    class="css-view-175oi2r r-backgroundColor-1niwhzg r-backgroundPosition-vvn4in r-backgroundRepeat-u6sd8q r-bottom-1p0dtai r-height-1pi2tsx r-left-1d2f490 r-position-u8s1d r-right-zchlnj r-top-ipm5af r-width-13qz1uu r-zIndex-1wyyakw r-backgroundSize-4gszlv"
  />
</div>
`;

exports[`components/Image prop "source" is correctly updated only when loaded if defaultSource provided 1`] = `
<div
  class="css-view-175oi2r r-flexBasis-1mlwlqe r-overflow-1udh08x r-zIndex-417010"
>
  <div
    class="css-view-175oi2r r-backgroundColor-1niwhzg r-backgroundPosition-vvn4in r-backgroundRepeat-u6sd8q r-bottom-1p0dtai r-height-1pi2tsx r-left-1d2f490 r-position-u8s1d r-right-zchlnj r-top-ipm5af r-width-13qz1uu r-zIndex-1wyyakw r-backgroundSize-4gszlv"
    style="background-image: url(https://testing.com/preview.jpg);"
  />
  <img
    alt=""
    class="css-accessibilityImage-9pa8cd"
    draggable="false"
    src="https://testing.com/preview.jpg"
  />
</div>
`;

exports[`components/Image prop "source" is correctly updated only when loaded if defaultSource provided 2`] = `
<div
  class="css-view-175oi2r r-flexBasis-1mlwlqe r-overflow-1udh08x r-zIndex-417010"
>
  <div
    class="css-view-175oi2r r-backgroundColor-1niwhzg r-backgroundPosition-vvn4in r-backgroundRepeat-u6sd8q r-bottom-1p0dtai r-height-1pi2tsx r-left-1d2f490 r-position-u8s1d r-right-zchlnj r-top-ipm5af r-width-13qz1uu r-zIndex-1wyyakw r-backgroundSize-4gszlv"
    style="background-image: url(https://testing.com/fullSize.jpg);"
  />
  <img
    alt=""
    class="css-accessibilityImage-9pa8cd"
    draggable="false"
    src="https://testing.com/fullSize.jpg"
  />
</div>
`;

exports[`components/Image prop "source" is correctly updated when missing in initial render 1`] = `
<div
  class="css-view-175oi2r r-flexBasis-1mlwlqe r-overflow-1udh08x r-zIndex-417010"
>
  <div
    class="css-view-175oi2r r-backgroundColor-1niwhzg r-backgroundPosition-vvn4in r-backgroundRepeat-u6sd8q r-bottom-1p0dtai r-height-1pi2tsx r-left-1d2f490 r-position-u8s1d r-right-zchlnj r-top-ipm5af r-width-13qz1uu r-zIndex-1wyyakw r-backgroundSize-4gszlv"
    style="background-image: url(https://testing.com/img.jpg);"
  />
  <img
    alt=""
    class="css-accessibilityImage-9pa8cd"
    draggable="false"
    src="https://testing.com/img.jpg"
  />
</div>
`;

exports[`components/Image prop "source" is not set immediately if the image has not already been loaded 1`] = `
<div
  class="css-view-175oi2r r-flexBasis-1mlwlqe r-overflow-1udh08x r-zIndex-417010"
>
  <div
    class="css-view-175oi2r r-backgroundColor-1niwhzg r-backgroundPosition-vvn4in r-backgroundRepeat-u6sd8q r-bottom-1p0dtai r-height-1pi2tsx r-left-1d2f490 r-position-u8s1d r-right-zchlnj r-top-ipm5af r-width-13qz1uu r-zIndex-1wyyakw r-backgroundSize-4gszlv"
    style="background-image: url(https://google.com/favicon.ico);"
  />
  <img
    alt=""
    class="css-accessibilityImage-9pa8cd"
    draggable="false"
    src="https://google.com/favicon.ico"
  />
</div>
`;

exports[`components/Image prop "source" is set immediately if the image has already been loaded 1`] = `
<div
  class="css-view-175oi2r r-flexBasis-1mlwlqe r-overflow-1udh08x r-zIndex-417010"
>
  <div
    class="css-view-175oi2r r-backgroundColor-1niwhzg r-backgroundPosition-vvn4in r-backgroundRepeat-u6sd8q r-bottom-1p0dtai r-height-1pi2tsx r-left-1d2f490 r-position-u8s1d r-right-zchlnj r-top-ipm5af r-width-13qz1uu r-zIndex-1wyyakw r-backgroundSize-4gszlv"
    style="background-image: url(https://google.com/favicon.ico);"
  />
  <img
    alt=""
    class="css-accessibilityImage-9pa8cd"
    draggable="false"
    src="https://google.com/favicon.ico"
  />
</div>
`;

exports[`components/Image prop "source" is set immediately if the image has already been loaded 2`] = `
<div
  class="css-view-175oi2r r-flexBasis-1mlwlqe r-overflow-1udh08x r-zIndex-417010"
>
  <div
    class="css-view-175oi2r r-backgroundColor-1niwhzg r-backgroundPosition-vvn4in r-backgroundRepeat-u6sd8q r-bottom-1p0dtai r-height-1pi2tsx r-left-1d2f490 r-position-u8s1d r-right-zchlnj r-top-ipm5af r-width-13qz1uu r-zIndex-1wyyakw r-backgroundSize-4gszlv"
    style="background-image: url(https://twitter.com/favicon.ico);"
  />
  <img
    alt=""
    class="css-accessibilityImage-9pa8cd"
    draggable="false"
    src="https://twitter.com/favicon.ico"
  />
</div>
`;

exports[`components/Image prop "source" is set immediately if the image was preloaded 1`] = `
<div
  class="css-view-175oi2r r-flexBasis-1mlwlqe r-overflow-1udh08x r-zIndex-417010"
>
  <div
    class="css-view-175oi2r r-backgroundColor-1niwhzg r-backgroundPosition-vvn4in r-backgroundRepeat-u6sd8q r-bottom-1p0dtai r-height-1pi2tsx r-left-1d2f490 r-position-u8s1d r-right-zchlnj r-top-ipm5af r-width-13qz1uu r-zIndex-1wyyakw r-backgroundSize-4gszlv"
    style="background-image: url(https://yahoo.com/favicon.ico);"
  />
  <img
    alt=""
    class="css-accessibilityImage-9pa8cd"
    draggable="false"
    src="https://yahoo.com/favicon.ico"
  />
</div>
`;

exports[`components/Image prop "style" removes other unsupported View styles 1`] = `
<div
  class="css-view-175oi2r r-flexBasis-1mlwlqe r-overflow-1udh08x r-zIndex-417010"
>
  <div
    class="css-view-175oi2r r-backgroundColor-1niwhzg r-backgroundPosition-vvn4in r-backgroundRepeat-u6sd8q r-bottom-1p0dtai r-height-1pi2tsx r-left-1d2f490 r-position-u8s1d r-right-zchlnj r-top-ipm5af r-width-13qz1uu r-zIndex-1wyyakw r-backgroundSize-4gszlv"
    style="filter: url(#tint-57);"
  />
  <svg
    style="position: absolute; height: 0px; visibility: hidden; width: 0px;"
  >
    <defs>
      <filter
        id="tint-57"
      >
        <feflood
          flood-color="blue"
        />
        <fecomposite
          in2="SourceAlpha"
          operator="atop"
        />
      </filter>
    </defs>
  </svg>
</div>
`;

exports[`components/Image prop "style" supports "resizeMode" property 1`] = `
<div
  class="css-view-175oi2r r-flexBasis-1mlwlqe r-overflow-1udh08x r-zIndex-417010"
>
  <div
    class="css-view-175oi2r r-backgroundColor-1niwhzg r-backgroundPosition-vvn4in r-backgroundRepeat-u6sd8q r-bottom-1p0dtai r-height-1pi2tsx r-left-1d2f490 r-position-u8s1d r-right-zchlnj r-top-ipm5af r-width-13qz1uu r-zIndex-1wyyakw r-backgroundSize-ehq7j7"
  />
</div>
`;

exports[`components/Image prop "style" supports "shadow" properties (convert to filter) 1`] = `
<div
  class="css-view-175oi2r r-flexBasis-1mlwlqe r-overflow-1udh08x r-zIndex-417010"
>
  <div
    class="css-view-175oi2r r-backgroundColor-1niwhzg r-backgroundPosition-vvn4in r-backgroundRepeat-u6sd8q r-bottom-1p0dtai r-height-1pi2tsx r-left-1d2f490 r-position-u8s1d r-right-zchlnj r-top-ipm5af r-width-13qz1uu r-zIndex-1wyyakw r-backgroundSize-4gszlv"
    style="filter: drop-shadow(1px 1px 0px rgba(255,0,0,1.00));"
  />
</div>
`;

exports[`components/Image prop "style" supports "tintcolor" property (convert to filter) 1`] = `
<div
  class="css-view-175oi2r r-flexBasis-1mlwlqe r-overflow-1udh08x r-zIndex-417010"
>
  <div
    class="css-view-175oi2r r-backgroundColor-1niwhzg r-backgroundPosition-vvn4in r-backgroundRepeat-u6sd8q r-bottom-1p0dtai r-height-1pi2tsx r-left-1d2f490 r-position-u8s1d r-right-zchlnj r-top-ipm5af r-width-13qz1uu r-zIndex-1wyyakw r-backgroundSize-4gszlv"
    style="background-image: url(https://google.com/favicon.ico); filter: url(#tint-56);"
  />
  <img
    alt=""
    class="css-accessibilityImage-9pa8cd"
    draggable="false"
    src="https://google.com/favicon.ico"
  />
  <svg
    style="position: absolute; height: 0px; visibility: hidden; width: 0px;"
  >
    <defs>
      <filter
        id="tint-56"
      >
        <feflood
          flood-color="red"
        />
        <fecomposite
          in2="SourceAlpha"
          operator="atop"
        />
      </filter>
    </defs>
  </svg>
</div>
`;

exports[`components/Image prop "testID" 1`] = `
<div
  class="css-view-175oi2r r-flexBasis-1mlwlqe r-overflow-1udh08x r-zIndex-417010"
  data-testid="testID"
>
  <div
    class="css-view-175oi2r r-backgroundColor-1niwhzg r-backgroundPosition-vvn4in r-backgroundRepeat-u6sd8q r-bottom-1p0dtai r-height-1pi2tsx r-left-1d2f490 r-position-u8s1d r-right-zchlnj r-top-ipm5af r-width-13qz1uu r-zIndex-1wyyakw r-backgroundSize-4gszlv"
  />
</div>
`;
